app/frontend/src/components/charts/utils/shared.ts

```diff
@@ -20,3 +20,36 @@ export const calculatePrecision = (
 
   return null // FUll precision
 }
+
```

Comment:
Can we include 0-value-filling for tokens missing in some of the entries? I remember you've already done this for some of the old charts and this is the most likely reason for UNI display issue we saw on the demo.